crates/router/src/connector/powertranz.rs

```diff
@@ -283,19 +278,14 @@ impl
         &self,
         req: &types::PaymentsCompleteAuthorizeRouterData,
         _connectors: &settings::Connectors,
-    ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
         let redirect_payload: powertranz::RedirectResponsePayload = req
             .request
             .get_redirect_response_payload()?
             .parse_value("PowerTranz RedirectResponsePayload")
             .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
         let spi_token = format!(r#""{}""#, redirect_payload.spi_token);
-        let powertranz_req =
-            types::RequestBody::log_and_get_request_body(&spi_token, |spi_token| {
-                Ok(spi_token.to_string())
-            })
-            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
-        Ok(Some(powertranz_req))
+        Ok(RequestContent::Json(Box::new(spi_token)))
```

Comment:
request body here is `text/plain`. Let's introduce one more type in `RequestContent` which handle text/plain body